runners/samza/src/test/java/org/apache/beam/runners/samza/renderer/PipelineJsonRendererTest.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.samza.renderer;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.Sum;
+import org.apache.beam.sdk.transforms.windowing.FixedWindows;
+import org.apache.beam.sdk.transforms.windowing.Window;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.TimestampedValue;
+import org.joda.time.Duration;
+import org.joda.time.Instant;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests for {@link PipelineJsonRenderer}. */
+@RunWith(JUnit4.class)
+public class PipelineJsonRendererTest {
+
+  static {
+    System.setProperty("beamUseDummyRunner", Boolean.TRUE.toString());
+  }
```

Comment:
why do we need this?

Reply:
This is because it requires a runner to be initialized for the test. It will otherwise throw an exception saying "No Runner was specified and the DirectRunner was not found on the classpath".

Reply:
Is there any specific reason to use TestPipeline instead of Pipeline here? There are several examples of using regular Pipeline in unit tests [1].

In addition, if we need to use TestPipeline, I would expect we init it with [2]
```
TestPipeline.fromOptions(
          PipelineOptionsFactory.fromArgs("--runner=TestSamzaRunner").create());
```


[1] https://github.com/apache/beam/blob/master/runners/samza/src/test/java/org/apache/beam/runners/samza/translation/ConfigGeneratorTest.java#L64
[2] https://github.com/apache/beam/blob/master/runners/samza/src/test/java/org/apache/beam/runners/samza/runtime/SamzaStoreStateInternalsTest.java#L84